File: Problems/Algorithms/Median of Two Sorted Arrays/medianTwoArrays.kt
class Solution {
    fun findMedianSortedArrays(nums1: IntArray, nums2: IntArray): Double {
        var m = nums1.size
        var n = nums2.size
        val mergedNums = IntArray(m+n)
        var k = 0
        var i = 0
        var j = 0
        
        while (i < m && j < n) {
            if (nums1[i] < nums2[j]) {
                mergedNums[k++] = nums1[i]
                i++
            } else {
                mergedNums[k++] = nums2[j]
                j++
            }
        }
        
        while (i < m) {
            mergedNums[k++] = nums1[i]
            i++
        }
        while (j < n) {
            mergedNums[k++] = nums2[j]
            j++
        }
        
        if ((m+n) % 2 == 0) {
            return (mergedNums[(m+n)/2-1] + mergedNums[(m+n)/2]).toDouble() / 2
        } else {
            return mergedNums[(m+n)/2].toDouble()
        }
    }
}

File: Problems/Algorithms/4. Median of Two Sorted Arrays/median-two-arrays.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int pointer1 = 0, pointer2 = 0;

    int getMin(vector<int>& nums1, vector<int>& nums2) {
        int m = nums1.size(), n = nums2.size();

        if (pointer1 < m && pointer2 < n) {
            return nums1[pointer1] < nums2[pointer2] ? nums1[pointer1++] : nums2[pointer2++];
        } else if (pointer1 < m) {
            return nums1[pointer1++];
        } else if (pointer2 < n) {
            return nums2[pointer2++];
        }

        return -1;
    }

    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        int m = nums1.size(), n = nums2.size();

        if ((m + n) % 2 == 0) {
            for (int i = 0; i < (m + n) / 2 - 1; ++i) {
                int _ = getMin(nums1, nums2);
            }
            return (double)(getMin(nums1, nums2) + getMin(nums1, nums2)) / 2;
        } else {
            for (int i = 0; i < (m + n) / 2; ++i) {
                int _ = getMin(nums1, nums2);
            }
            return getMin(nums1, nums2);
        }
            
        return -1;
    }
};
